### ISSUE TITLE
Crash under python <3.10: got Future <Future pending> attached to a different loop

### ISSUE THREAD

**Issue by rytilahti** [rytilahti/python-eq3bt#57]
Using Dockerfile:
```
FROM python:3.9

WORKDIR /app
RUN apt-get update && apt-get upgrade -y
RUN apt-get install -y bluez bluetooth

RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
ENV PATH="${PATH}:/root/.poetry/bin"

COPY . .
RUN poetry install

ENTRYPOINT ["poetry", "run", "eq3cli"]
```
executing the cli crashes on an eventloop related issue:
```
❯ docker run -v /var/run/dbus/:/var/run/dbus/:z eq3bt:latest --mac 00:1A:22:xx --debug
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:eq3bt.eq3btsmart:Querying the device..
DEBUG:eq3bt.bleakconnection:Trying to connect to 00:1A:22:xx
DEBUG:eq3bt.bleakconnection:Connected to 00:1A:22:xx
DEBUG:eq3bt.bleakconnection:Writing b'0316071311051f' to 1041
DEBUG:eq3bt.bleakconnection:Waiting for notifications for 1
DEBUG:eq3bt.bleakconnection:Got notification from 1057: b'020189000409'
DEBUG:eq3bt.eq3btsmart:Received notification from the device..
DEBUG:eq3bt.eq3btsmart:Got status: b'020189000409'
DEBUG:eq3bt.eq3btsmart:Parsed status: Container: 
    cmd = 2
    mode = Container: 
        AUTO = True
        MANUAL = True
        DST = True
        LOW_BATTERY = True
    valve = 0
    target_temp = 4.5
    away = None
    presets = None
DEBUG:eq3bt.eq3btsmart:Valve state:      0
DEBUG:eq3bt.eq3btsmart:Mode:             manual off dst low battery
DEBUG:eq3bt.eq3btsmart:Target temp:      4.5
DEBUG:eq3bt.eq3btsmart:Away end:         None
DEBUG:eq3bt.eq3btsmart:Window open temp: None
DEBUG:eq3bt.eq3btsmart:Window open time: None
DEBUG:eq3bt.eq3btsmart:Comfort temp:     None
DEBUG:eq3bt.eq3btsmart:Eco temp:         None
DEBUG:eq3bt.eq3btsmart:Temp offset:      None
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/root/.cache/pypoetry/virtualenvs/python-eq3bt-9TtSrW0h-py3.9/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/root/.cache/pypoetry/virtualenvs/python-eq3bt-9TtSrW0h-py3.9/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/root/.cache/pypoetry/virtualenvs/python-eq3bt-9TtSrW0h-py3.9/lib/python3.9/site-packages/click/core.py", line 1635, in invoke
    rv = super().invoke(ctx)
  File "/root/.cache/pypoetry/virtualenvs/python-eq3bt-9TtSrW0h-py3.9/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/.cache/pypoetry/virtualenvs/python-eq3bt-9TtSrW0h-py3.9/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/root/.cache/pypoetry/virtualenvs/python-eq3bt-9TtSrW0h-py3.9/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/app/eq3bt/eq3cli.py", line 47, in cli
    thermostat.update()
  File "/app/eq3bt/eq3btsmart.py", line 217, in update
    self._conn.make_request(PROP_WRITE_HANDLE, value)
  File "/app/eq3bt/bleakconnection.py", line 119, in make_request
    self._loop.run_until_complete(self.wait_for_response(timeout))
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/app/eq3bt/bleakconnection.py", line 101, in wait_for_response
    await asyncio.wait_for(self._notifyevent.wait(), timeout)
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 479, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.9/asyncio/locks.py", line 226, in wait
    await fut
RuntimeError: Task <Task pending name='Task-9' coro=<Event.wait() running at /usr/local/lib/python3.9/asyncio/locks.py:226> cb=[_release_waiter(<Future pendi...7e95b6640>()]>)() at /usr/local/lib/python3.9/asyncio/tasks.py:416]> got Future <Future pending> attached to a different loop
```

**Comment by rytilahti** [comment]
Seems to work on `python:3.10`, failing on 3.8 and 3.9 images.

**Comment by rytilahti** [comment]
With asyncio debug enabled, those unclosed loops are likely caused by the crash itself?
```
sys:1: ResourceWarning: unclosed <socket.socket fd=9, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0, raddr=/run/dbus/system_bus_socket>
ResourceWarning: Enable tracemalloc to get the object allocation traceback
sys:1: ResourceWarning: unclosed <socket.socket fd=10, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0, raddr=/run/dbus/system_bus_socket>
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/local/lib/python3.8/asyncio/base_events.py:654: ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=True>
DEBUG:asyncio:Close <_UnixSelectorEventLoop running=False closed=False debug=True>
```

**Comment by Tarkon33** [comment]
Is it possible to use the bluepy backend when I use it via library?
I couldn't find an option yet to do so like in the CLI with --backend

**Comment by rytilahti** [comment]
Yes, by passing `BTLEConnection` as the connection class to your Thermostat constructor:
```
from eq3bt import Thermostat
from eq3bt.connection import BTLEConnection
t = Thermostat("mac", connection_cls=BTLEConnection)
```

**Comment by jdemel** [comment]
@rytilahti I saw the same error when I tested your library. First: thanks for maintaining it.

A quick fix for my quick CLI test was to comment out line 119
https://github.com/rytilahti/python-eq3bt/blob/abe4208b5b4833ea2e58e57fa867528cea87d1ff/eq3bt/bleakconnection.py#L119

Unfortunately, I couldn't make the other backends work. This issue kind of indicates that multiple `asyncio.run` calls (or similar) get in the way in the end.
